vnet/conn_map.go

```diff
@@ -0,0 +1,135 @@
+package vnet
+
+import (
+	"fmt"
+	"net"
+)
+
+type udpConnMap struct {
+	portMap map[int][]*UDPConn
+}
+
+func newUDPConnMap() *udpConnMap {
+	return &udpConnMap{
+		portMap: map[int][]*UDPConn{},
+	}
+}
+
+func (m *udpConnMap) insert(conn *UDPConn) error {
+	udpAddr := conn.LocalAddr().(*net.UDPAddr)
+
+	// check if the port has a listener
+	conns, ok := m.portMap[udpAddr.Port]
+	if ok {
+		if udpAddr.IP.IsUnspecified() {
```

Comment:
sorry I don't understand this specific check here, what case are you covering here?

Reply:
Say, there is a socket listening on 192.168.0.2:1234.
When you try to listen on 0.0.0.0:1234, it would hit this statement.

When 0.0.0.0 is used as a local address to bind, it tries to bind all IP address your local machine. If there's any IP address already listening to the port, it will end up with 'addrress already in use' error.

Reply:
of course, right, that makes total sense, it's clear now, thanks!